src/main/java/org/jabref/gui/desktop/os/Linux.java

```diff
@@ -89,41 +90,61 @@ public void openFileWithApplication(String filePath, String application) throws
     public void openFolderAndSelectFile(Path filePath) throws IOException {
         String desktopSession = System.getenv("DESKTOP_SESSION");
 
-        String cmd = "xdg-open " + filePath.toAbsolutePath().getParent().toString(); // default command
+        String cmd = "xdg-open"; // default command
 
         if (desktopSession != null) {
             desktopSession = desktopSession.toLowerCase(Locale.ROOT);
             if (desktopSession.contains("gnome")) {
-                cmd = "nautilus --select " + filePath.toString().replace(" ", "\\ ");
+                cmd = "nautilus --select";
             } else if (desktopSession.contains("kde") || desktopSession.contains("plasma")) {
-                cmd = "dolphin --select " + filePath.toString().replace(" ", "\\ ");
+                cmd = "dolphin --select";
             } else if (desktopSession.contains("mate")) {
-                cmd = "caja --select " + filePath.toString().replace(" ", "\\ ");
+                cmd = "caja --select";
             } else if (desktopSession.contains("cinnamon")) {
-                cmd = "nemo --select " + filePath.toString().replace(" ", "\\ ");
+                cmd = "nemo"; // Although nemo is based on nautilus it does not support --select, it directly highlights the file
             }
         }
-        Runtime.getRuntime().exec(cmd);
+        ProcessBuilder processBuilder = new ProcessBuilder(cmd, filePath.toAbsolutePath().toString());
```

Comment:
This does not work for me (arch Linux with Kde plasma).
```suggestion
        Runtime.getRuntime().exec(cmd + " " + filePath.toString().replace(" ", "\\ "));
```
works, but I guess there is a reason you changed to ProcessBuilder.

Reply:
Also, are whitespaces the only thing that we have to worry about when passing the file parameter?

Reply:
https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/lang/Runtime.html#exec(java.lang.String,java.lang.String%5B%5D)

>Deprecated.
This method is error-prone and should not be used, the corresponding method [exec(String[], String[])](https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/lang/Runtime.html#exec(java.lang.String%5B%5D,java.lang.String%5B%5D)) or [ProcessBuilder](https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/lang/ProcessBuilder.html) should be used instead. The command string is broken into tokens using only whitespace characters. For an argument with an embedded space, such as a filename, this can cause problems as the token does not include the full filename.